dev/com.ibm.ws.classloading/src/com/ibm/ws/classloading/internal/ClassLoadingServiceImpl.java

```diff
@@ -728,6 +739,15 @@ public boolean isThreadContextClassLoader(ClassLoader cl) {
     @Override
     @Trivial // injected trace calls ProtectedDomain.toString() which requires privileged access
     public void setSharedLibraryProtectionDomains(Map<String, ProtectionDomain> protectionDomainMap) {
+        
+        if (tc.isDebugEnabled()) {
+            Tr.debug(tc, "protectionDomainMap.size = " + protectionDomainMap.size());
+            for (Map.Entry<String, ProtectionDomain> entry : protectionDomainMap.entrySet()) {
+                Tr.debug(tc, "Key (codeBase)= " + entry.getKey() + ", Value (protectionDomain) = " + entry.getValue());
+            }
+
+        }
+        
```

Comment:
Can we have method for this debug so I can be used by other classes?

Reply:
Comments have been added where needed, including all new classes introduced.